azure-iot-device/azure/iot/device/iothub/pipeline/iothub_pipeline.py

```diff
@@ -21,15 +21,25 @@
 )
 from azure.iot.device.iothub.auth.x509_authentication_provider import X509AuthenticationProvider
 
+from azure.iot.device.iothub.config import ConfigProvider
+
 logger = logging.getLogger(__name__)
 
 
 class IoTHubPipeline(object):
-    def __init__(self, auth_provider):
+    def __init__(self, auth_provider, **kwargs):
         """
         Constructor for instantiating a pipeline adapter object
         :param auth_provider: The authentication provider
+        :param protocol: The protocol for connecting to IoT Hub
         """
+        try:
+            self._config = ConfigProvider(**kwargs)
```

Comment:
>self._config = ConfigProvider(**kwargs) [](start = 12, length = 39)

Nice!  
I like how you used **kwargs on the public interface, but then used explicit keyword args in the ConfigProvider object.  This is a great balance between the freedom that **kwargs gives us and the checking we get with explicit argument names